### ISSUE TITLE
"Expected no line gap between" false positive

### ISSUE THREAD

**Issue by nelsonprsousa** [yannickcr/eslint-plugin-react#2774]
I believe one of the last [commits](https://github.com/yannickcr/eslint-plugin-react/commit/5029bd1b588c3f72251e13de190f9df7bd1390d8) on master brings a new unexpected error. For example:

```typescript
<Tab.Screen
    name={`Dia ${index + 1}`}
    component={DayTab}
    // eslint-disable-next-line react/no-array-index-key
    key={index}
/>
```

`error  Expected no line gap between “component” and “key”  react/jsx-props-no-multi-spaces`

@iiison / @ljharb  is this okay? In these scenarios, how should we avoid the error?

**Comment by iiison** [comment]
Hmm. We're iterating over only on JSX attributes. I guess we need to update the rule to look for other elements in the tag. what do you think @ljharb ?

**Comment by ljharb** [comment]
I would expect a comment to be ignored, so yes, that seems reasonable.
